### PR TITLE
Feature/improve required param support

### PR DESCRIPTION
**Pull request by rma-rripken**
Adding a new exception and some Controllers helpers for required query parameters.  If they aren't found the helpers will throw the new exception and ApiServlet message will make clear to user that a required query parameter was missing.

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/api/Controllers.java [hunk 1]
```diff
@@ -260,4 +265,55 @@ public static JooqDao.DeleteMethod getDeleteMethod(String input) {
...
     }
+
+    /**
+     * Returns the first matching query param or throws RequiredQueryParameterException.
+     * @param ctx Request Context
+     * @param name Query parameter name
+     * @return value of the parameter
+     * @throws RequiredQueryParameterException if the parameter is not found
+     */
+    public static String requiredParam(io.javalin.http.Context ctx, String name) {
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/api/Controllers.java:276]: We don't need to this now but might be good if this allowed the caller (not client) to set a more detail message than just the name.

If you agree, open an issue and we'll get to it later.

File: cwms-data-api/src/main/java/cwms/cda/api/TimeSeriesController.java [hunk 2]
```diff
@@ -272,21 +275,20 @@ protected TimeSeriesDao getTimeSeriesDao(DSLContext dsl) {
     @Override
     public void delete(Context ctx, @NotNull String timeseries) {
 
-        String office = ctx.queryParam(OFFICE);
+        String office = requiredParam(ctx, OFFICE);
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/api/TimeSeriesController.java:278]: I don't think this one is required as the office of each timeseries is embedded in the body's JSON or XML.

Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/api/TimeSeriesController.java:278]: Then again maybe it should be. Something to think about.

**Review by MikeNeilson — APPROVED** [review]

File: cwms-data-api/src/main/java/cwms/cda/api/Controllers.java (hunk 1 above)
Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/api/Controllers.java:276]: That is a good idea. #561

File: cwms-data-api/src/main/java/cwms/cda/api/TimeSeriesController.java (hunk 2 above)
Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/api/TimeSeriesController.java:278]: The delete doesn't take a body but I agree we need to look and think about which params are required and where.  I started to go thru all the params marked required and not all of them actually are (some of the offices we mark as required but then the dao treat null as *).  Its also possible that some of the parameters are required in v1 api but optional in v2.  Not sure exactly how to handle those cases.  I guess we just document them when we find them.

Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/api/TimeSeriesController.java:278]: #562

**Pull request merged**